### ISSUE TITLE
[source-klaviyo]: connector missing fields ahead of Klaviyo API v1 Deprecation on June 30th 

### ISSUE THREAD

**Issue by glevineLeap** [airbytehq/airbyte#39554]
### Connector Name

source-klaviyo

### Connector Version

2.7.0

### What step the error happened?

None

### Relevant information

The v1/v2 legacy APIs are [scheduled to be retired](https://developers.klaviyo.com/en/docs/api_versioning_and_deprecation_policy) by Klaviyo on June 30th, 2024. As such, all source versions below 2.0 will stop working at that time and anyone using an earlier version will be forced to migrate onto a 2.0+ version. 

Currently, there is still not parity in the data available between the two major versions. For example, this [slack thread](https://airbytehq.slack.com/archives/C021JANJ6TY/p1715110316760429) calls out that "the new API doesn't return profile data or profile ID with event's stream data" and this requires calling another endpoint that Airbyte doesn't currently include.

Happy to contribute if someone can point me in the right direction.

### Contribute

- [X] Yes, I want to contribute

**Comment by marcosmarxm** [comment]
@glevineLeap the API revision the Klaviyo connector uses today is `2023-10-15` which has plans to be deprecated only in 2025.

https://github.com/airbytehq/airbyte/blob/23fd80aa2a3a6f8834192e6d9a018f2fec95fc49/airbyte-integrations/connectors/source-klaviyo/source_klaviyo/streams.py#L27

**Comment by marcosmarxm** [comment]
@glevineLeap it is confusing to me what is the goal of this issue. Are some field missing considering the current version, can you point the Klaviyo documentation to the endpoint and fields?

**Comment by glevineLeap** [comment]
> @glevineLeap the API revision the Klaviyo connector uses today is `2023-10-15` which has plans to be deprecated only in 2025.

@marcosmarxm Correct, but the current version is still not at data parity with the 1.x versions of the connector. So, _this_ version of the connector

> @glevineLeap it is confusing to me what is the goal of this issue. Are some field missing considering the current version, can you point the Klaviyo documentation to the endpoint and fields?

Specifically, in our case we don't have a way under the new structure to link events with flows/campaigns. There's an [included](https://developers.klaviyo.com/en/reference/get_events) field in Get Events that's not currently in the Airbyte schema.

**Comment by glevineLeap** [comment]
There are other relevant fields (for example [Campaigns](https://developers.klaviyo.com/en/reference/get_campaigns) no longer includes the number of recipients and theres a separate [Get Campaign Recipient Estimation](https://developers.klaviyo.com/en/reference/get_campaign_recipient_estimation) endpoint), but the event-flow/campaign tie is the only thing we don't have a workaround for right now.

**Comment by keligijus** [comment]
@marcosmarxm, [we're seeing the connector using version 2023-02-22](https://github.com/airbytehq/airbyte/issues/50941#issue-2769450140), which is to expire 1,5 months. Would you be able to look into this?

**Comment by andresTavioTJ** [comment]
Any update here? Revision 2023-02-22 is now retired
